client/src/ProtectedRoutes/ProtectedRoutes.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Navigate,
+} from 'react-router-dom';
+import { useAuth } from '../Context/AuthContext';
+
+export function ProtectedAdmin({ children }) {
+  const { user } = useAuth();
+  if (!user.user) {
+    return <Navigate to="/dashboard" replace />;
+  }
+  return children;
+}
+export function ProtectedUser({ children }) {
+  const { user } = useAuth();
+  if (user.user.role === 0) {
+    return <Navigate to="/" replace />;
+  }
+  return children;
+}
+export function ProtectedEngineer({ children }) {
+  const { user } = useAuth();
+  if (user.user.role === 2) {
```

Comment:
as I told you it's not readable to keep the roles as numbers. you can define constants in the useAuth hook to make it more readable

Reply:
```suggestion
  if (user.role === ENG_ROLE) {
```